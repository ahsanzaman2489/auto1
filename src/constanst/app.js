export const DEFAULT_PAGE_LIMIT = 10;
export const HOST = "http://localhost:";
export const HOST_PORT = 3001;
export const HEADERS = {"Access-Control-Allow-Origin":"*"};