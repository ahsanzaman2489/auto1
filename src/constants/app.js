export const HOST = "http://localhost:";
export const HOST_PORT = 3001;
export const NO_DATA = 'Opps...! No results found with this query';
export const NO_FAVOURITE = 'If you like this car, click the button and save it in your collection of favourite items.';
export const FAVOURITE = 'This Car is your favourite';
export const FOOTER_TEXT = `${new Date().getFullYear()} Â© AUTO1 Group`;