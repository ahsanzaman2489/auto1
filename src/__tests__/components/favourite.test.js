import React from 'react';
import FavouriteComponent from '../../components/favourite';
import {NO_FAVOURITE, FAVOURITE} from '../../constants/app.js';
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";

const setup = (props = {}, state = null) => {
    const defaultProps = {
        stockNumber: "10172",
        addToFavourite: jest.fn(),
        isFavourite: false,
        removeFromFavourite: jest.fn()
    };
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<FavouriteComponent {...setUpProps}/>);
    const tree = () => renderer.create(<FavouriteComponent {...setUpProps}/>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps,tree}
};

describe("Favourite component", () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });
    it('Should render correct text and button if component is not favourite', () => {
        const {wrapper, props} = setup();
        expect(wrapper.find("p").text()).toBe(NO_FAVOURITE);
        wrapper.find("button").simulate('click');
        expect(props.addToFavourite).toHaveBeenCalled();
        expect(props.addToFavourite).toHaveBeenCalledTimes(1);
    });

    it('Should render correct and button text if component is favourite', () => {
        const {wrapper, props} = setup({isFavourite: true});

        expect(wrapper.find("p").text()).toBe(FAVOURITE);
        wrapper.find("button").simulate('click');

        expect(props.removeFromFavourite).toHaveBeenCalled();
        expect(props.removeFromFavourite).toHaveBeenCalledTimes(1);
    });

});


