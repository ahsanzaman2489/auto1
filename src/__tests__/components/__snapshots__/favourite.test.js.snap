// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite component should match with snapshot 1`] = `
<div
  className="add-favourite"
>
  <p>
    If you like this car, click the button and save it in your collection of favourite items.
  </p>
  <button
    className="float-right btnfixedbtm"
    onClick={[Function]}
  >
    save
  </button>
</div>
`;
